src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java

```diff
@@ -85,6 +85,7 @@ public class GitHubPluginConfig extends GlobalConfiguration {
 
     public GitHubPluginConfig() {
         getConfigFile().getXStream().alias("github-server-config", GitHubServerConfig.class);
+        getConfigFile().getXStream().alias(GITHUB_PLUGIN_CONFIGURATION_ID, GitHubPluginConfig.class);
```

Comment:
Though the actual problem is that `load()` is called in the constructor it looks like a too involved fix to make that right. Instead all aliases already have a central point where they should be made.
```suggestion
        GitHubPlugin.addXStreamAliases();
```
And add `alias("github-server-config", GitHubServerConfig.class)` to `org.jenkinsci.plugins.github.migration.Migrator#enableAliases` instead.